File: app/src/main/java/com/sahibindencourseproject/api/model/DailyForecastResponse.java

package com.sahibindencourseproject.api.model;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;

public class DailyForecastResponse {

    @SerializedName("city")
    @Expose
    private City city;
    @SerializedName("cod")
    @Expose
    private String cod;
    @SerializedName("message")
    @Expose
    private Double message;
    @SerializedName("cnt")
    @Expose
    private Integer cnt;
    @SerializedName("list")
    @Expose
    private ArrayList<WeatherItem> weatherItem = null;

    public City getCity() {
        return city;
    }

    public void setCity(City city) {
        this.city = city;
    }

    public String getCod() {
        return cod;
    }

    public void setCod(String cod) {
        this.cod = cod;
    }

    public Double getMessage() {
        return message;
    }

    public void setMessage(Double message) {
        this.message = message;
    }

    public Integer getCnt() {
        return cnt;
    }

    public void setCnt(Integer cnt) {
        this.cnt = cnt;
    }

    public ArrayList<WeatherItem> getWeatherItem() {
        return weatherItem;
    }

    public void setWeatherItem(ArrayList<WeatherItem> weatherItem) {
        this.weatherItem = weatherItem;
    }

}

File: app/src/main/java/com/sahibindencourseproject/api/model/DailyForecastResponse.kt
package com.sahibindencourseproject.api.model

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

import java.util.ArrayList

class DailyForecastResponse() : Parcelable {

    @SerializedName("city")
    @Expose
    var city: City? = null
    @SerializedName("cod")
    @Expose
    var cod: String? = null
    @SerializedName("message")
    @Expose
    var message: Double? = null
    @SerializedName("cnt")
    @Expose
    var cnt: Int? = null
    @SerializedName("list")
    @Expose
    var weatherItem: ArrayList<WeatherItem>? = null

    constructor(parcel: Parcel) : this() {
        city = parcel.readParcelable(City::class.java.classLoader)
        cod = parcel.readString()
        message = parcel.readValue(Double::class.java.classLoader) as? Double
        cnt = parcel.readValue(Int::class.java.classLoader) as? Int
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeParcelable(city, flags)
        parcel.writeString(cod)
        parcel.writeValue(message)
        parcel.writeValue(cnt)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<DailyForecastResponse> {
        override fun createFromParcel(parcel: Parcel): DailyForecastResponse {
            return DailyForecastResponse(parcel)
        }

        override fun newArray(size: Int): Array<DailyForecastResponse?> {
            return arrayOfNulls(size)
        }
    }

}
